docs/cow-protocol/tutorials/widget/widget.md

```diff
@@ -69,6 +69,62 @@ const params: CowSwapWidgetParams = {
 createCowSwapWidget(widgetContainer, { params })
 ```
 
+### Flexible config
+
+Both `bps` and `recipient` can be set for different chains and different trade types (swap/limit/advanced). 
+
+Bellow you can see the `partnerFee` configuration variations:
+
+```typescript
+import {PartnerFee, SupportedChainId, TradeType} from '@cowprotocol/widget-lib'
+
+// The fee is 1% for all trades on all chains
+const a: PartnerFee = {
+    bps: 100,
+    recipient: '0x0000000000000000000000000000000000000000'
+}
+
+// Per network and per trade type
+const b: PartnerFee = {
+    bps: {
+        [SupportedChainId.MAINNET]: {
+            [TradeType.SWAP]: 100, [TradeType.LIMIT]: 50, [TradeType.ADVANCED]: 30,
+        },
+        [SupportedChainId.ARBITRUM_ONE]: {
+            [TradeType.SWAP]: 100, [TradeType.LIMIT]: 50, [TradeType.ADVANCED]: 30,
+        },
+        [SupportedChainId.GNOSIS_CHAIN]: {
+            [TradeType.SWAP]: 100, [TradeType.LIMIT]: 50, [TradeType.ADVANCED]: 30,
+        },
+        [SupportedChainId.SEPOLIA]: {
+            [TradeType.SWAP]: 100, [TradeType.LIMIT]: 50, [TradeType.ADVANCED]: 30,
+        },
+    },
+    recipient: '0x0000000000000000000000000000000000000000',
+}
+
+// Per trade type and per network
+const c: PartnerFee = {
+    bps: 100,
+    recipient: {
+        [TradeType.SWAP]: {
+            [SupportedChainId.MAINNET]: '0x...a', [SupportedChainId.ARBITRUM_ONE]: '0x...b',
+            [SupportedChainId.GNOSIS_CHAIN]: '0x...c', [SupportedChainId.SEPOLIA]: '0x...d',
+        },
+        [TradeType.LIMIT]: {
+            [SupportedChainId.MAINNET]: '0x...e', [SupportedChainId.ARBITRUM_ONE]: '0x...f',
+            [SupportedChainId.GNOSIS_CHAIN]: '0x...g', [SupportedChainId.SEPOLIA]: '0x...h',
+        },
+        [TradeType.ADVANCED]: {
+            [SupportedChainId.MAINNET]: '0x...j', [SupportedChainId.ARBITRUM_ONE]: '0x...i',
+            [SupportedChainId.GNOSIS_CHAIN]: '0x...k', [SupportedChainId.SEPOLIA]: '0x...l',
+        },
+    }
+}
+```
+
+See [FlexibleConfig](https://github.com/cowprotocol/cowswap/blob/develop/libs/widget-lib/src/types.ts) type for more information. 
```

Comment:
I think we should add a note on how to do even more flexible ones using the listeners. I believe this config will cover 90% of integrations, but is still nice to point out that we have even a more fine level of granularity allowing you to set configurations based on flow


This actually makes me think if we should allow to provide a configuration at some point using a function just to make it even easier.

How it works is, the user defines a function that receive all relevant data: network, trade params, flow, etc.
They return the partnerFee configuration

Our widget library would remember the last partnerFee reported to the widget, and if it changed, it would send the new one. This way, we don't notify changes all the time, but let the user have a very flexible logic.

Reply:
https://github.com/cowprotocol/docs/pull/392/commits/744105cde8a3e3469b99f26ebcdec9284a3dd7ac